lib/ansible/module_utils/basic.py

```diff
@@ -2018,8 +2013,8 @@ def _return_formatted(self, kwargs):
             else:
                 self.warn(kwargs['warnings'])
 
-        if self._warnings:
-            kwargs['warnings'] = self._warnings
+        if warnings._global_warnings:
```

Comment:
if you are externally accessing it, it should not be a private var, or you might want to consider a read only method instead

Reply:
That may not be a bad idea. Ideally updating these values could only be done via `warn()` and `deprecate()`.